Update to ONVM 19.07

archit-p opened the pull request:
This pull request updates the onvm-snort repository to run with onvm 19.07.

Changes were made in _libdaq_ code and configure scripts. A gitignore file was added to ensure clean commits.

ratnadeepb committed:
Update Install.md

ratnadeepb committed:
Added two simple CPU timing counters to daq_netvm.c

The snort NF will now print out processing time for every batch of packets and the related packets per second rate

Commit:
Automating compile and install

patching-Makefile.sh has been modified to accept default values. This file is now called from inside setenv.sh. setenv.sh sets up all initial enviornment variables assuming defaults. The variables in this file can be changed if defaults are not used. The setenv.sh script is run from inside install.sh. This script essentially, sets up all environment variables, installs all packages and builds all the tools.

archit-p committed:
Updated API to to onvm 19.07

archit-p committed:
Added DPDK flags to daq-2.0.6/configure.ac

Flags needed to compile daq-2.0.6 with DPDK 18.11

archit-p committed:
Added build scripts

daq-2.0.6/build.sh - automates daq build
snort-2.9.8.3/build.sh - automates snort build

archit-p committed:
Added .gitignore

.gitignore contains common patterns to ignore for C projects.

archit-p committed:
Code cleanup

Removed extra print statements, added comments to daq_netvm.c

dennisafa commented on daq-2.0.6/Makefile:
```diff
@@ -195,24 +206,24 @@ distuninstallcheck_listfiles = find . -type f -print
   | sed 's|^\./|$(prefix)/|' | grep -v '$(infodir)/dir$$'
 distcleancheck_listfiles = find . -type f -print
-ACLOCAL = ${SHELL} /users/graceliu/git/onvm-snort/daq-2.0.6/missing aclocal-1.14
+ACLOCAL = ${SHELL} /home/student/sources/onvm-snort/daq-2.0.6/missing aclocal-1.15
```
Should `/home/student/sources` be set to a shell variable so that it is correct for any given user?

archit-p replied:
I believe the Makefile is auto-generated using the configure script which contains the shell variable so that it is correct for any given user

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -148,26 +163,38 @@ static int parse_args(char *inputstring, char **argv)
+/* need a sig handler for cleanup */
+void sig_handler(int sig) {
+    if (sig == SIGINT || sig == SIGTERM) {
+        //printf("Received user termination");
```
nit: Remove commented print statement

archit-p replied:
Updated

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -148,26 +163,38 @@ static int parse_args(char *inputstring, char **argv)
+    /* these are needed for new standards */
+    struct onvm_nf_local_ctx *nf_local_ctx;
 
+    //printf("->netvm_daq_initialize()\n");
```
Nit: Rm commented print statement. Or surround in a #if debug variable.

archit-p replied:
Updated

archit-p commented:
Oops looks like I forgot to update the openNetVM version in the repository. Would it be better to use a openNetVM as a submodule here?

archit-p committed:
Added openNetVM 19.07 as a git submodule

archit-p committed:
Updated install.sh and setenv.sh script

Removed patching-makefile.sh step, changed variable names

twood02 commented:
Yes, using a submodule would be ideal.  Let us know when it is ready for us to take a look again.  Thanks @archit-p !

archit-p commented:
Hi,

I've updated the repository to include openNetVM 19.07 as a submodule. Have also updated the installation instructions in install.md, and installation script in install.sh. I'd really appreciate it if someone could follow these instructions and try a fresh installation!

archit-p commented:
Hello team, 

Just a gentle reminder on the open pull request. Please provide suggestions on how to take this forward.

twood02 commented:
Thanks @archit-p .  @dennisafa is going to take a look but he may not get to it for another week because of class work.

archit-p commented on Install.md:
````diff
@@ -53,28 +38,28 @@ This guide helps you build and install Snort.
     ```
 
-2. Clean the build. 
+3. Clean the build. 
````
Yes, looks like it. 

When I run according to the current instructions, `make clean` fails. Shall I move step 3.5 to 3.3?

archit-p commented:
@dennisafa is the current code building alright?

dennisafa commented:
Hi Archit, i’ve run into a few issues compiling the code. I’ll get to it
this Thursday - i have a few deadlines before then.

On Tue, Oct 22, 2019 at 12:35 PM Archit Pandey <notifications@github.com>
wrote:

> @dennisafa <https://github.com/dennisafa> is the current code building
> alright?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sdnfv/onvm-snort/pull/8?email_source=notifications&email_token=AH3EIZUMNB3OYYI32PDN2NLQP4TUXA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEB6MN4I#issuecomment-545048305>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH3EIZWAO4AGJWFQ77BMZ7LQP4TUXANCNFSM4I76KWHA>
> .
>

archit-p commented:
Hi dennis,

No problems, you could share the issues here

dennisafa commented on Install.md:
````diff
@@ -87,19 +68,25 @@ This guide helps you build and install Snort.
+    autoheader
+    automake -a
+    ```
+3. Run the configuration script.
````
Running into issues on this step:
<img width="408" alt="Screen Shot 2019-10-24 at 10 52 06 PM" src="https://user-images.githubusercontent.com/32916582/67539732-1399c680-f6b1-11e9-9b27-f57fd2589140.png">

dennisafa replied:
I think the issue is that the $PATH variable needs to be initialized to point to the `daq-modules-config` file. This file is found in `/daq-2.0.6/os-daq-modules`

archit-p commented:
That's strange, I didn't need to update my $PATH. Could you share the snort
`config.log` file? It might contain what the error is exactly.

On Fri, 25 Oct 2019, 8:37 am Dennis Afanasev, <notifications@github.com>
wrote:

> *@dennisafa* commented on this pull request.
> ------------------------------
>
> In Install.md
> <https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338869726>:
>
> >      ```sh
> -    ./configure --enable-sourcefire
> +    aclocal
> +    autoconf
> +    autoheader
> +    automake -a
> +    ```
> +3. Run the configuration script.
>
> I think the issue is that the $PATH variable needs to be initialized to
> point to the daq-modules-config file. This file is found in
> /daq-2.0.6/os-daq-modules
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sdnfv/onvm-snort/pull/8?email_source=notifications&email_token=AHMM4LEO5PPRTH5LVSR32NLQQJPINA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJF6B4I#discussion_r338869726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHMM4LE5ACXAWGBTY3XAC3LQQJPINANCNFSM4I76KWHA>
> .
>

dennisafa commented:
> That's strange, I didn't need to update my $PATH. Could you share the snort `config.log` file? It might contain what the error is exactly.
> […](#)
> On Fri, 25 Oct 2019, 8:37 am Dennis Afanasev, ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In Install.md <[#8 (comment)](https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338869726)>: > ```sh - ./configure --enable-sourcefire + aclocal + autoconf + autoheader + automake -a + ``` +3. Run the configuration script. I think the issue is that the $PATH variable needs to be initialized to point to the daq-modules-config file. This file is found in /daq-2.0.6/os-daq-modules — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#8?email_source=notifications&email_token=AHMM4LEO5PPRTH5LVSR32NLQQJPINA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJF6B4I#discussion_r338869726>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHMM4LE5ACXAWGBTY3XAC3LQQJPINANCNFSM4I76KWHA> .

<img width="1276" alt="Screen Shot 2019-10-24 at 11 21 13 PM" src="https://user-images.githubusercontent.com/32916582/67540859-01ba2280-f6b5-11e9-992d-2af22fd63aba.png">
Here is the relevant config.log section. Looks like there are issue with -lonvm and -lonvmhelper

archit-p commented:
Looks like $DPDK_TARGET is not being interpreted correctly.

Could you try `echo $DPDK_TARGET` and if nothing pops up, `export
$DPDK_TARGET=x86_64-native-linuxapp-gcc` or your corresponding architecture.

On a side note, I believe I should remove the use of $DPDK_TARGET and stick
to $RTE_TARGET in daq-2.0.6/configure.ac. WDYT?

On Fri, 25 Oct 2019, 8:52 am Dennis Afanasev, <notifications@github.com>
wrote:

> That's strange, I didn't need to update my $PATH. Could you share the
> snort config.log file? It might contain what the error is exactly.
> … <#m_306845524257962216_>
> On Fri, 25 Oct 2019, 8:37 am Dennis Afanasev, *@*.*> wrote: @.**
> commented on this pull request. ------------------------------ In
> Install.md <#8 (comment)
> <https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338869726>>: > sh
> - ./configure --enable-sourcefire + aclocal + autoconf + autoheader +
> automake -a + +3. Run the configuration script. I think the issue is that
> the $PATH variable needs to be initialized to point to the
> daq-modules-config file. This file is found in /daq-2.0.6/os-daq-modules —
> You are receiving this because you were mentioned. Reply to this email
> directly, view it on GitHub <#8
> <https://github.com/sdnfv/onvm-snort/pull/8>?email_source=notifications&email_token=AHMM4LEO5PPRTH5LVSR32NLQQJPINA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJF6B4I#discussion_r338869726>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AHMM4LE5ACXAWGBTY3XAC3LQQJPINANCNFSM4I76KWHA
> .
>
> [image: Screen Shot 2019-10-24 at 11 21 13 PM]
> <https://user-images.githubusercontent.com/32916582/67540859-01ba2280-f6b5-11e9-992d-2af22fd63aba.png>
> Here is the relevant config.log section. Looks like there are issue with
> -lonvm and -lonvmhelper
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sdnfv/onvm-snort/pull/8?email_source=notifications&email_token=AHMM4LDHD6AWV3HYZOG34JLQQJQ53A5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHB2HQ#issuecomment-546184478>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHMM4LDBWT62T5AI2VI5BKDQQJQ53ANCNFSM4I76KWHA>
> .
>

dennisafa commented:
> Looks like $DPDK_TARGET is not being interpreted correctly. Could you try `echo $DPDK_TARGET` and if nothing pops up, `export $DPDK_TARGET=x86_64-native-linuxapp-gcc` or your corresponding architecture. On a side note, I believe I should remove the use of $DPDK_TARGET and stick to $RTE_TARGET in daq-2.0.6/configure.ac. WDYT? On Fri, 25 Oct 2019, 8:52 am Dennis Afanasev, <notifications@github.com> wrote:
> […](#)
> That's strange, I didn't need to update my $PATH. Could you share the snort config.log file? It might contain what the error is exactly. … <#m_306845524257962216_> On Fri, 25 Oct 2019, 8:37 am Dennis Afanasev, *@*.*> wrote: @.** commented on this pull request. ------------------------------ In Install.md <#8 (comment) <[#8 (comment)](https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338869726)>>: > sh - ./configure --enable-sourcefire + aclocal + autoconf + autoheader + automake -a + +3. Run the configuration script. I think the issue is that the $PATH variable needs to be initialized to point to the daq-modules-config file. This file is found in /daq-2.0.6/os-daq-modules — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#8 <#8>?email_source=notifications&email_token=AHMM4LEO5PPRTH5LVSR32NLQQJPINA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJF6B4I#discussion_r338869726>, or unsubscribe https://github.com/notifications/unsubscribe-auth/AHMM4LE5ACXAWGBTY3XAC3LQQJPINANCNFSM4I76KWHA . [image: Screen Shot 2019-10-24 at 11 21 13 PM] <https://user-images.githubusercontent.com/32916582/67540859-01ba2280-f6b5-11e9-992d-2af22fd63aba.png> Here is the relevant config.log section. Looks like there are issue with -lonvm and -lonvmhelper — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#8?email_source=notifications&email_token=AHMM4LDHD6AWV3HYZOG34JLQQJQ53A5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECHB2HQ#issuecomment-546184478>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHMM4LDBWT62T5AI2VI5BKDQQJQ53ANCNFSM4I76KWHA> .

Yes, let's stick with using $RTE_SDK as that is part of the openNetVM install guide. Thank you.

archit-p commented:
Sure, will do that!

I remember the old guide using two variables `RTE_TARGET` and `DPDK_TARGET` with the same value.

archit-p committed:
Changed DPDK_TARGET to RTE_TARGET

Maintains consistency with openNetVM 19.07

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -186,21 +211,44 @@ static int netvm_daq_initialize(const DAQ_Config_t *config, void **ctxt_ptr, cha
+    onvm_nflib_start_signal_handler(nf_local_ctx, sig_handler);
+    /*function table not needed */
+
+    /*
```
Nit: Remove unnecessary commented code

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -186,21 +211,44 @@ static int netvm_daq_initialize(const DAQ_Config_t *config, void **ctxt_ptr, cha
     printf("netvm init done\n");
-    /* Complete onvm handshake */
-    onvm_nflib_nf_ready(nf_info);
+    /* not necessary honestly
```
Nit: Remove commets

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -213,13 +261,23 @@ static int netvm_daq_initialize(const DAQ_Config_t *config, void **ctxt_ptr, cha
-    netvmc->rx_ring = onvm_nflib_get_rx_ring(nf_info);
-    netvmc->tx_stats = onvm_nflib_get_tx_stats(nf_info);
+
+    /* these functions do not exist anymore
```
Nit: Remove commented code

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -273,7 +331,7 @@ static int netvm_daq_start(void *handle)
     netvm_daq_reset_stats(handle);
 
-    //nf_info->status = NF_RUNNING;
+    //nf->status = NF_RUNNING;
```
Nit: Remove commented code

dennisafa commented on daq-2.0.6/os-daq-modules/daq_netvm.c:
```diff
@@ -401,15 +472,24 @@ static int netvm_daq_acquire(void *handle, int cnt, DAQ_Analysis_Func_t callback
+	    nf->stats.tx += tx_batch_size;
 	}
     }
     //printf("<-netvm_daq_acquire() - count\n");
```
Remove commented out print statements

archit-p commented:
There are extra print statements in several places. I believe, they were used when debugging daq_netvm.c.

archit-p commented:
There are too many commits in my repository right now, I wish to squash or fixup most of them. 

Could you let me know once you approve of the changes? I'll make a force push with modified history.

dennisafa commented on Install.md:
````diff
@@ -87,19 +68,25 @@ This guide helps you build and install Snort.
    ```
 
-4. Navigate to the src folder of snort and Make snort.
+5. Navigate to the src folder of snort and Make snort.
````
Please add a step asking the user to run `autoreconf -ivf` in case of a version mismatch error, then to do step 3 again. 
Also, I am getting this issue: 
<img width="1280" alt="Screen Shot 2019-10-24 at 11 57 43 PM" src="https://user-images.githubusercontent.com/32916582/67542278-2369d880-f6ba-11e9-80b7-dcceb6cb1cf6.png">

Could you elaborate on -ldpdk a bit more?

archit-p replied:
Yes, looks like I forgot to add `autoreconf -fvi`. Thanks for pointing that out!

dennisafa replied:
Hi Archit, please add this step to the install guide. Thank you.

archit-p commented:
-ldpdk is found in $RTE_SDK/$RTE_TARGET/lib

From the screenshot it looks to me like your $RTE_SDK variable is still set
to the dpdk-16.11 directory. You might need to set it again.

On Fri, 25 Oct 2019, 9:28 am Dennis Afanasev, <notifications@github.com>
wrote:

> *@dennisafa* commented on this pull request.
> ------------------------------
>
> In Install.md
> <https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338877707>:
>
> >     ```sh
>     cd ../
>     ./patching-Makefile.sh
>     ```
>
> -4. Navigate to the src folder of snort and Make snort.
> +5. Navigate to the src folder of snort and Make snort.
>
> Please add a step asking the user to run autoreconf -ivf in case of a
> version mismatch error, then to do step 3 again.
> Also, I am getting this issue:
> [image: Screen Shot 2019-10-24 at 11 57 43 PM]
> <https://user-images.githubusercontent.com/32916582/67542278-2369d880-f6ba-11e9-80b7-dcceb6cb1cf6.png>
>
> Could you elaborate on -ldpdk a bit more?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sdnfv/onvm-snort/pull/8?email_source=notifications&email_token=AHMM4LFSHMT2VGLOO7RECDLQQJVHTA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJGANEA#pullrequestreview-306972304>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHMM4LGLPKGSNW5DDL46A6DQQJVHTANCNFSM4I76KWHA>
> .
>

dennisafa commented:
> -ldpdk is found in $RTE_SDK/$RTE_TARGET/lib From the screenshot it looks to me like your $RTE_SDK variable is still set to the dpdk-16.11 directory. You might need to set it again.
> […](#)
> On Fri, 25 Oct 2019, 9:28 am Dennis Afanasev, ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In Install.md <[#8 (comment)](https://github.com/sdnfv/onvm-snort/pull/8#discussion_r338877707)>: > ```sh cd ../ ./patching-Makefile.sh ``` -4. Navigate to the src folder of snort and Make snort. +5. Navigate to the src folder of snort and Make snort. Please add a step asking the user to run autoreconf -ivf in case of a version mismatch error, then to do step 3 again. Also, I am getting this issue: [image: Screen Shot 2019-10-24 at 11 57 43 PM] <https://user-images.githubusercontent.com/32916582/67542278-2369d880-f6ba-11e9-80b7-dcceb6cb1cf6.png> Could you elaborate on -ldpdk a bit more? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#8?email_source=notifications&email_token=AHMM4LFSHMT2VGLOO7RECDLQQJVHTA5CNFSM4I76KWHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJGANEA#pullrequestreview-306972304>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHMM4LGLPKGSNW5DDL46A6DQQJVHTANCNFSM4I76KWHA> .

It looks like my $RTE_SDK output is correct (set to onvm-snort/openNetVM/dpdk) I will look into it.

archit-p commented:
Might be helpful to try running 
```
sudo ldconfig
```
once DAQ is done installing.

archit-p commented:
Hi, any updates on the build status?

dennisafa commented:
> Hi, any updates on the build status?
> 
> You could try running:
> 
> ```
> aclocal
> autoconf 
> automake -a
> autoheader
> ./configure --enable-sourcefire --enable-static --disabled-shared
> ```
> 
> and skipping the patching step. I'm unsure of whether the patching step is needed.

I will try this today - thanks for the communication.

archit-p commented:
Hello, 

I've run a clean build multiple times locally, would appreciate if someone with a different setup could verify the build. 

We could use some more eyes to the pull request @twood02 @ratnadeepb!

dennisafa commented on Install.md:
````diff
@@ -87,19 +68,25 @@ This guide helps you build and install Snort.
     ```
 
-3. Create Patch of Makefille      
+4. Create Patch of Makefille. This part is critical for the next steps. The defaults are blank. Please type in the answers manually.
````
Did you input anything specific here? Without the patch, the build completes. If you didn't enter anything specific, lets remove this step (traditional snort install doesn't do this)

archit-p replied:
Yes, I do not know the rationale behind this. Perhaps someone else could expand.

dennisafa commented on Install.md:
````diff
@@ -121,19 +108,28 @@ This guide helps you build and install Snort.
     ```
+2. Add snort to path (change /opt/snort if the install path is different)
+    ```
+    export PATH=$PATH:/opt/snort/bin"
````
Remove " at the end. Should be `export PATH=$PATH:/opt/snort/bin`

dennisafa commented on Install.md:
````diff
@@ -121,19 +108,28 @@ This guide helps you build and install Snort.
+    If the above does not work then try:
+    ```
+    which snort
+    sudo `which snort` -A console -Q -c /etc/snort/snort.conf -i dpdk0:dpdk1 -N --alert-before-pass --daq-var netvm_args="-l 5 -n 3 --proc-type=secondary -- -r 1 -- -d 4"
````
Running into an issue here:
<img width="625" alt="Screen Shot 2019-10-31 at 2 42 10 PM" src="https://user-images.githubusercontent.com/32916582/67976591-c6f04700-fbec-11e9-8a3a-ad5a6cc65d6e.png">

Did you run into something similar?

archit-p replied:
Did not come accross this issue.. I'll look into what might be causing this

archit-p replied:
Looks like when using --disable-shared, snort does not build the dynamic_preprocessors. I've edited the snort.conf to disable these dynamic_preprocessors.

archit-p committed:
Update Install.md

Steps to build onvm-snort with openNetVM 19.07

archit-p committed:
Disabled dynamic preprocessors

Using --disable-shared while building snort, does not build
dynamic preprocessors.

archit-p commented:
In order to fix this error, please pull the latest changes, and repeat `Step 5 Configure and run openNetVM-snort` in install.md.

dennisafa commented:
> In order to fix this error, please pull the latest changes, and repeat `Step 5 Configure and run openNetVM-snort` in install.md.

Thanks archit, I'll run a fresh build today.

dennisafa commented on Install.md:
```diff
@@ -3,79 +3,61 @@
     aclocal
     autoconf
     autoheader
+    autoreconf -fvi
```
this should be `autoreconf -ivf` the current order causes an error in running `make -j7 ` in which tokdef.h is not found

dennisafa commented on Install.md:
```diff
@@ -121,19 +103,28 @@ This guide helps you build and install Snort.
-    cd openNetVM-dev/onvm
-    ./go.sh 0,1,2,3,4 3 -v 0x7f000000000
+    cd openNetVM/onvm
+    ./go.sh 0,1,2,3 3 0xF0 -v 0x7f000000000 -s stdout
```
Please change `-v 0x7f000000000`  to `-a 0x7f000000000`

dennisafa commented on Install.md:
```diff
@@ -87,23 +69,23 @@ This guide helps you build and install Snort.
+    aclocal
+    autoconf
+    autoheader
+    autoreconf -fvi
```
Same here, `autoreconf -ivf`

dennisafa commented on Install.md:
````diff
@@ -87,23 +69,23 @@ This guide helps you build and install Snort.
-
+3. Run the configuration script.
+    ```sh
+    ./configure --enable-sourcefire --enable-static
````
Let's keep this as `./configure --enable-sourcefire --enable-static --disable-shared` I ran into issues with --disable-shared left out, DPDK complained about conflicting links.

dennisafa commented:
I managed to get a working build running with a few modifications to the install guide (see comments above) 
After you change those, I'll approve the PR. Thank you!

archit-p commented:
Hi @dennisafa 

Thanks a lot for reviewing the changes. I've updated the pull request according to your suggestions above.

archit-p committed:
Added Archit Pandey to README.md

Updated installation instructions

archit-p commented:
@dennisafa I've merged commits into a more manageable number. You could merge these if the changes are approved. Thanks.

archit-p commented:
Hi, does this merge request require anymore changes?

dennisafa commented:
> Hi, does this merge request require anymore changes?

Hi Archit, I ran into a few issues when re-building on a fresh instance. I've had a lot of work pile up so I haven't gotten around to figuring out the exact issue quite yet. I will retry and submit a change to the install guide by this weekend. Thanks for your patience!

dennisafa commented on Install.md:
```diff
@@ -3,79 +3,54 @@
     aclocal
     autoconf
     autoheader
+    autoreconf -ivf
```
Put `autoreconf -ivf` before aclocal.

dennisafa commented on Install.md:
```diff
@@ -87,28 +62,26 @@ This guide helps you build and install Snort.
+    aclocal
+    autoconf
+    autoheader
+    autoreconf -ivf
```
Same here

dennisafa reviewed (changes requested):
Hi Archit, thanks for being patient. I managed to get snort working by switching a couple commands in the install guide. Could you change those, and then squash the unnecessary commits? After that I will definitely approve. 
<img width="1277" alt="Screen Shot 2019-12-07 at 5 01 54 PM" src="https://user-images.githubusercontent.com/32916582/70381131-59ce8080-1913-11ea-9d5f-fdc3fd6431a0.png">

dennisafa commented:
Hey @archit-p let me know if you have any questions about the comments!

archit-p commented:
Hi, apologies for being away for so long. I have made the requested changes and updated the pull request! Thanks for reviewing!

archit-p commented:
Hi @twood02! Could you look at the proposed changes?

twood02 commented:
Thanks @archit-p! I'll be discussing this with Dennis tomorrow and expect to merge it after that.  We already have other students who are interested in using your code so your contribution is greatly appreciated!
